wire/fromwire.c

```diff
@@ -392,6 +392,19 @@ struct bitcoin_tx_output *fromwire_bitcoin_tx_output(const tal_t *ctx,
 	return output;
 }
 
+struct witscript *fromwire_witscript(const tal_t *ctx, const u8 **cursor, size_t *max)
+{
+	struct witscript *retval = tal(ctx, struct witscript);
+	u16 len = fromwire_u16(cursor, max);
+	if (len == 0) {
+		retval->ptr = NULL;
+	} else {
+		retval->ptr = tal_arr(retval, u8, len);
+		fromwire_u8_array(cursor, max, retval->ptr, len);
+	}
+	return retval;
```

Comment:
For consistency, I think this should return NULL (just like you allocate witscripts in a tx as NULL), rather than a struct containing a NULL ptr.

Something like:

```
struct witscript *retval;
u16 len = fromwire_u16(cursor, max);

if (!len)
    return NULL;

retval = tal(ctx, struct witscript);
retval->ptr = tal_arr(retval, u8, len);
fromwire_u8_array(cursor, max, retval->ptr, len);
return retval;
```

Reply:
Fixed.